### DISCUSSION TITLE
Support for Android

### DISCUSSION THREAD

**Discussion by fayaz-modz** [raphamorim/rio#393] [Ideas]
Any intentions to support for android?

If not cam you give me a roadmap of porting this to android?

**Comment by raphamorim** [comment]
Hey there @fayaz-modz ,

Thanks for the issue. It isn't in a near roadmap because Rio still have some work left regarding WebAssembly but I think that shouldn't be much trouble to adjust the renderer and other dependencies like winit to android.

For reference, could you describe a bit more the use cases or similar terminals to android?

**Comment by fayaz-modz** [comment]
As you might already know that android is based on linux. And arm processors these days are getting more and more powerful.

Consider this, I'm using termux, a full blown root fs like environment with a good package manager, I'm using termux on my android tablet with chroot with ubuntu arm, which runs at native speed. I've been able to successfully use this environment for flutter (a crossplatform gui development patform) and I'm developling android apps using this.

Although android ndk is not yet compiled for arm but I'm able to make most of the apps without any issues. The terminal runs smooth.

I'm ready for any contributions, I can find time on my week days. It looks like a great project, if will walk me though the file structure and how to get up running, I can fix bugs and possibly bring new feature when kept in guidence. Thank you.

**Comment by raphamorim** [comment]
That would be lovely @fayaz-modz ,

Please feel free to join the discord channel (link is in the rio website)

Some links that could be useful to port Rio to Android.

- https://github.com/rust-windowing/winit#android
- https://github.com/BrainiumLLC/cargo-mobile
